Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -296,27 +298,52 @@ Refer to the framework documentation for examples of session timeout configurati
 
 #### Dynamic Analysis
 
-You can use dynamic analysis to verify that authorization is consistently enforced on all remote endpoints. First, manually or automatically crawl the application to make sure that all privileged actions and data are secure and to determine whether a valid session ID is required. Record the requests in your proxy.
+To verify if a session timeout is implemented, proxy your requests through an interception proxy:
+
+1. Log into the application.
+2. Perform a couple of operations that require authentication.
+3. Leave the session idle until it expires.
```

Comment:
This is if you know how long the session timeout would be. I propose we change it to the following:

> 2. Access a resource that requires authentication, typically a request for private information belonging to your account.
> 3. Try to access the data after an increasing number of 5-minute delays has passed (5, 10, 15, ...).
> 4. Once the resource is no longer available, you will know the session timeout.
>
> If you can still access the resources after a timeout that is too long, this test case fails and the session timeout is not invalidating the session on server-side.
>
> Tip: When using the Burp proxy, you can use the [Session Timeout Test extension](https://portswigger.net/bappstore/c4bfd29882974712a1d69c6d8f05874e "Session Timeout Test") to automate this test.

(And add that URL in the tools references ?)

Reply:
Thanks for rephrasing. I copied it in, but I got rid of "that is too long". 

If you can still access the resources after a timeout, this test case fails and the session timeout is not invalidating the session on server-side.

The sentence is now easier to read.